image_to_table/image_with_word_boxes.py

```diff
@@ -1,146 +1,87 @@
-
-    box_ys = list(zip(starts, stops))
-    row_boxes = [
-        Box(
-            lower_right_corner=Point(x=max_x, y=stop),
-            lower_left_corner=Point(x=0, y=stop),
-            upper_right_corner=Point(x=max_x, y=start),
-            upper_left_corner=Point(x=0, y=start),
-        )
-        for (start, stop) in box_ys
-    ]
-    return row_boxes
-
-
-def make_column_boxes(x_counts, max_y):
-    inside = False
-    starts = []
-    stops = []
-
-    for x, n in enumerate(x_counts):
-        if n > 0 and not inside:
-            starts.append(x - 1)
-            inside = True
-        if n == 0 and inside:
-            stops.append(x)
-            inside = False
-    if inside:
-        stops.append(x)
-
-    box_xs = list(zip(starts, stops))
-    column_boxes = [
-        Box(
-            lower_right_corner=Point(y=max_y, x=stop),
-            lower_left_corner=Point(y=0, x=stop),
-            upper_right_corner=Point(y=max_y, x=start),
-            upper_left_corner=Point(y=0, x=start),
-        )
-        for (start, stop) in box_xs
-    ]
-    return column_boxes
-
-
-def show_all_boxes_intersecting(filename, row_boxes, column_boxes):
-    for row_box in row_boxes:
-        for column_box in column_boxes:
-            show_image_with_word_boxes(filename, [row_box, column_box])
-
-
-def get_row(row_box, column_boxes, original_boxes):
-    new_row = [get_cell_in_row(column_box, original_boxes, row_box) for column_box in column_boxes]
-    return new_row
-
-
-def get_cell_in_row(column_box, original_boxes, row_box):
-    row_column = []
-    for i, original_box in enumerate(original_boxes):
-        if original_box.is_inside_box(column_box) and original_box.is_inside_box(row_box):
-            row_column.append(original_box.text)
-    return " ".join(row_column)
-
-
-def extract_table_from_image(filename):
+
+def merge_sorted_text_boxes(boxes: List[TextBox]) -> TextBox:
+    bounding_rect = boxes[0].rect or boxes[-1].rect
+    description = " ".join(map(lambda x: x.text, boxes))
+
+    return TextBox(description, bounding_rect)
+
+
+def extract_table_from_image(filename: str, placement: List[int]) -> List[List[TextBox]]:
     image = cv2.imread(filename)
     height, width, _ = image.shape
     original_boxes = detect_text(filename)
     boxes = original_boxes
-    for box in original_boxes:
-        box.fill("black")
-    x_counts = [len(boxes_in_x(boxes, x)) for x in range(width)]
-    boxes_in_ys = [boxes_in_y(boxes, y) for y in range(height)]
-    y_counts = [len(l) for l in boxes_in_ys]
-    row_boxes = make_row_boxes(y_counts, width)
-    column_boxes = make_column_boxes(x_counts, height)
-    """
-    for box in row_boxes + column_boxes:
-        box.plot(color="black")
-    """
-    plt.axis("off")
-    plt.savefig("table_b.png")
-    plt.show()
 
-    table = [get_row(row_box, column_boxes, original_boxes) for row_box in row_boxes]
+    # Sort boxes by row
+    sorted_boxes = sorted(boxes, key=lambda box: box.rect.y)
```

Comment:
Suggestion: to make the comment unnecessary, could change the variable name to `boxes_sorted_by_row` or `sorted_boxes_by_row`